build/release.sh

```diff
@@ -417,24 +421,32 @@ elif [ $PLATFORM_ID -eq 12 ] || [ $PLATFORM_ID -eq 13 ] || [ $PLATFORM_ID -eq 14
         SUFFIX="-m"
     fi
     USE_SWD_JTAG="n"
-
-    # Compose, echo and execute the `make` command
-    MAKE_COMMAND="make -s clean all PLATFORM_ID=$PLATFORM_ID COMPILE_LTO=n DEBUG_BUILD=$DEBUG_BUILD MODULAR=$MODULAR USE_SWD_JTAG=$USE_SWD_JTAG USE_SWD=n"
-    if [ "$MODULAR" = "n" ]; then
-        MAKE_COMMAND+=" APP=tinker-serial1-debugging"
-    else
-        MAKE_COMMAND+=" APP=tinker"
-    fi
-    echo $MAKE_COMMAND
-    eval $MAKE_COMMAND
-
-    # Migrate file(s) into output interface
-    if [ "$MODULAR" = "n" ]; then
-        release_binary "tinker-serial1-debugging" "tinker-serial1-debugging" "$SUFFIX" "$DEBUG_BUILD" "$USE_SWD_JTAG"
-    else
-        release_binary "system-part1" "system-part1" "$SUFFIX" "$DEBUG_BUILD" "$USE_SWD_JTAG"
-        release_binary "user-part" "tinker" "$SUFFIX" "$DEBUG_BUILD" "$USE_SWD_JTAG"
-    fi
+    declare -a debugApps=("tinker-serial1-debugging" "tinker-serial-debugging")
+
+    for debugApp in ${debugApps[@]}; do
+        # Compose, echo and execute the `make` command
+        MAKE_COMMAND="make -s clean all PLATFORM_ID=$PLATFORM_ID COMPILE_LTO=n DEBUG_BUILD=$DEBUG_BUILD MODULAR=$MODULAR USE_SWD_JTAG=$USE_SWD_JTAG USE_SWD=n"
+        if [ "$MODULAR" = "n" ]; then
+            MAKE_COMMAND+=" APP=$debugApp"
+        else
+            MAKE_COMMAND+=" APP=tinker"
+        fi
+        echo $MAKE_COMMAND
+        eval $MAKE_COMMAND
+
+        # Migrate file(s) into output interface
+        if [ "$MODULAR" = "n" ]; then
+            release_binary "$debugApp" "$debugApp" "$SUFFIX" "$DEBUG_BUILD" "$USE_SWD_JTAG"
+        else
+            release_binary "system-part1" "system-part1" "$SUFFIX" "$DEBUG_BUILD" "$USE_SWD_JTAG"
+            release_binary "user-part" "tinker" "$SUFFIX" "$DEBUG_BUILD" "$USE_SWD_JTAG"
+        fi
+
+        # Keep as explicit last step
```

Comment:
You should mention why this needs to be the last step (i.e. Only non-modular builds should loop).

Reply:
I had that comment originally when I had the break just after the `release_binary "user-part" ...` step, but then thought it should really be made the explicit last step in case more code needs to be added.  Because the code is so simple `if MODULAR=y, break` I thought it should be self documenting.

Reply:
To that point, it is obvious what's happening only because there is so little code.

My preference would be to break immediately. No reason to write code for things that don't exist, and when the code becomes more complex it may still require a more complex solution.

Reply:
I would say your suggestion is craftier, although it's not easier to understand that `declare -a apps=("tinker")` doesn't do anything besides cause the for loop to be executed once, and act as a break.  Oddly enough, this might be more clear `declare -a apps=("modular-y-this-is-not-used")`, or maybe changing `release_binary "user-part" "tinker"` to `release_binary "user-part" "$app"`.

Reply:
I would disagree, "declare apps" followed by "for app in apps" practically reads like English.

Again, I'm not blocking the PR, I'm just offering my opinion at this point.

Reply:
I merged our suggestions together :)